ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -581,11 +649,28 @@ CURLcode HttpOperation::Setup()
   curl_error_message_[0] = '\0';
   curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_ERRORBUFFER, curl_error_message_);
 
+// Support for custom debug function callback was added in version 7.9.6 so we guard against
+// exposing the default CURL output by keeping verbosity always disabled in lower versions.
+#if LIBCURL_VERSION_NUM < CURL_VERSION_BITS(7, 9, 6)
   rc = SetCurlLongOption(CURLOPT_VERBOSE, 0L);
   if (rc != CURLE_OK)
   {
     return rc;
   }
+#else
+  rc = SetCurlLongOption(CURLOPT_VERBOSE, static_cast<long>(is_log_enabled_ || kEnableCurlLogging));
```

Comment:
Two concerns here:

* not sure if casting a boolean to a long will always give 0L / 1L, so a plain if statement to set a verbose local variable will be safer
* should it be `is_log_enabled && kEnableCurlLogging` instead ?

Reply:
Regarding the explicit cast, it looked fairly safe given that the CURL API seems to only care about the value being 0 (disabled) or otherwise. Will update to a good ol' if statement, as requested.

As for the logic, my impression was that we wanted to have the runtime drive this behavior regardless of what is set at compile-time, but if omitting `console_debug`, it would still output logs provided `WITH_CURL_LOGGING` option was set.

If, what is ultimately needed is that the compile-time flag dictate the runtime behavior, then I guess it would need to be AND'ed, instead.

Let me know what is the preferred approach here and I will push all the required changes at the same time.

Reply:
Regarding the OR (`||`) or AND (`&&`) logic for the test:

* using the patch as written (OR), one can set the default at compile time to log details, without having to change the `console_debug` attribute for each exporter, which requires an application change.
* this sounds "easier" to use, but I think there is a twist.

Assume an application has a trace exporter and a metric exporter, both using OTLP HTTP to different endpoints.

By setting `WITH_CURL_LOGGING=ON`, both exporter are now traced, always. There is no way to turn off any of this tracing, even when debugging is no longer needed.

For this reason, I think the AND logic is better.

For people who wants to trace everything, it forces the application to use something like:

```
OtlpHttpExporterOptions opt;
#ifdef MY_OWN_DEBUG_FLAG
opt.console_debug = true; // or read from an application specific configuration
#else
opt.console_debug = false;
#endif
```

so in effect, the logic that was present in http_operation_curl.cc is moved to the application space, where the application (not opentelemetry-cpp) decides if/how it wants tracing to be done.

Yes, it is a code change for the application, to benefit from the feature.

However, this also allows to trace a metric exporter independently of a trace exporter, which I think is much more convenient to debug things, compared to the OR logic, especially when exporters points to different endpoints.

And when debugging is no longer needed ("or read from an application specific configuration"), tracing can be turned off, without having to recompile.

Please change the logic to AND then.

Reply:
That is a convincing argument, thanks for explaining in detail. Changes made to AND the flags and get rid of the cast.